common/odp/odp.go

```diff
@@ -84,3 +84,35 @@ func AddDatapathInterface(dpname string, ifname string) error {
 	_, err = dp.CreateVport(odp.NewNetdevVportSpec(ifname))
 	return err
 }
+
+func CreateVxlanVport(dpname string, name string, portno uint16) (odp.VportID, string, error) {
```

Comment:
I've tried refactoring it in two ways:

* Reuse `CreateDummyVxlanVport` (https://github.com/weaveworks/weave/pull/2410/commits/19f52c9486831ae8b1ad42ad0cd75d111c520bb7)
* Move the MTU check to `common/odp.CreateDatapath` (https://github.com/weaveworks/weave/pull/2410/commits/4090b5f0a55095dbb086ee6663daf10fe8fa0656)

The latter does not create multiple netlink sockets for the checks. Also, it reuses the vxlan-vport created in the support check. However, it introduces the `common/net` artifact (due to circular deps).